### ISSUE TITLE
Visual Eval Pipeline: Phase 1 Baseline Script & Scaffolding

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#116]
## Summary

Create the Phase 1 baseline generation pipeline for visual UMAP evaluation. This establishes the Python script that generates synthetic datasets, fits Python UMAP, exports fuzzy k-NN graphs for Rust consumption, and produces baseline comparison plots.

This is the foundation that all subsequent visual eval work builds on.

## Wave

**Wave 1** — no dependencies, can be implemented in parallel with the Rust Export Tool issue.

### 1. Scaffolding

- Add `tests/visual_eval/output/` to `.gitignore` (append to existing file)
- Create `tests/visual_eval/output/.gitkeep` (ensures directory exists in git)
- Create `tests/visual_eval/README.md` documenting:
  - Purpose of the visual eval pipeline
  - Two-phase workflow (Phase 1: baseline, Phase 2: compare)
  - How to run each phase
  - What outputs are generated
  - Python environment setup (reference `tests/environment.yml`)

### 2. Phase 1 Python Script

Create `tests/visual_eval/generate_umap_comparisons.py` with `--phase baseline` support.

#### CLI Interface

```bash
python tests/visual_eval/generate_umap_comparisons.py --phase baseline [--dataset DATASET_NAME] [--output-dir DIR]
```

- `--phase`: Required. `baseline` for Phase 1.
- `--dataset`: Optional. Run a single dataset (e.g., `blobs_1000`). Default: run all Tier 1 datasets.
- `--output-dir`: Optional. Default: `tests/visual_eval/output/`.

#### Tier 1 Synthetic Datasets

Implement dataset generators using `sklearn.datasets`:

| Dataset Name | Generator | Parameters | n | features | Purpose |
|---|---|---|---|---|---|
| `blobs_1000` | `make_blobs` | 5 centers | 1000 | 10 | Well-separated Gaussians (easy case) |
| `circles_1000` | `make_circles` | noise=0.05, factor=0.5 | 1000 | 2 | Manifold-aware init (inner/outer rings) |
| `swiss_roll_2000` | `make_swiss_roll` | — | 2000 | 3 | Classic manifold learning test |
| `two_moons_1000` | `make_moons` | noise=0.05 | 1000 | 2 | Non-convex cluster separation |
| `blobs_hd_2000` | `make_blobs` | 10 centers | 2000 | 50 | High-dimensional stress test |

For swiss_roll, color by position along the roll (the `t` parameter from `make_swiss_roll`). Discretize into 5 bins for label coloring.

#### UMAP Fitting

Use consistent parameters for all datasets:

```python
n_neighbors = 15
min_dist = 0.1
n_components = 2
metric = "euclidean"
random_state = 42
```

Use `umap-learn` directly (not sklearn wrapper).

#### Python Spectral Init Extraction

Get the pre-SGD spectral init coordinates using:

```python
from umap.spectral import spectral_layout
init_coords = spectral_layout(
    mapper.graph_, n_components=2,
    random_state=np.random.RandomState(42)
)
```

This calls Python's own spectral initialization on the same graph.

#### Eigenvalue Spectrum

Compute the 10 smallest eigenvalues of the normalized Laplacian:

```python
from scipy.sparse.linalg import eigsh
from scipy.sparse import eye

# Build symmetric normalized Laplacian from UMAP's graph
graph = mapper.graph_
degree = np.array(graph.sum(axis=1)).flatten()
D_inv_sqrt = diags(1.0 / np.sqrt(np.maximum(degree, 1e-10)))
L = eye(graph.shape[0]) - D_inv_sqrt @ graph @ D_inv_sqrt
eigenvalues, _ = eigsh(L, k=10, which='SM')
```

#### Exports (per dataset)

All saved to `tests/visual_eval/output/`:

| File | Format | Contents |
|---|---|---|
| `{name}_graph.npz` | CSR sparse (f32 data, i32 indices/indptr/shape) | Fuzzy k-NN graph for Rust |
| `{name}_py_spectral.npy` | ndarray f64, shape (n, 2) | Python spectral init coords (pre-SGD) |
| `{name}_py_final.npy` | ndarray f32, shape (n, 2) | Python UMAP final embedding (post-SGD) |
| `{name}_labels.npy` | ndarray int, shape (n,) | Ground truth labels |
| `{name}_baseline.png` | PNG image | 2x2 baseline plot |

**Graph export format** must match `tests/fixtures/*/step5a_pruned.npz` so Rust's `load_sparse_csr_f32_u32` can read it:

```python
np.savez(
    path,
    data=graph.data.astype(np.float32),
    indices=graph.indices.astype(np.int32),
    indptr=graph.indptr.astype(np.int32),
    shape=np.array(graph.shape, dtype=np.int32),
    format=b"csr",
)
```

#### Baseline Plot (2x2 grid per dataset)

Saved as `{name}_baseline.png`:

```
┌───────────────────────────┬───────────────────────────┐
│  Python Spectral Init     │  Python UMAP (full SGD)   │
│  (pre-SGD coordinates)    │  (final embedding)        │
│  colored by true label    │  colored by true label    │
├───────────────────────────┼───────────────────────────┤
│  Eigenvalue Spectrum      │  Quantitative Metrics     │
│  (bar chart, first 10     │  (text panel)             │
│   eigenvalues)            │                           │
└───────────────────────────┴───────────────────────────┘
```

**Scatter plot style:**
- Point size: 1-3, alpha: 0.5 for large datasets (n > 500)
- Colormap: `tab10` or `tab20` for cluster labels
- No axis labels (UMAP coordinates are arbitrary)
- Title with dataset name and init strategy

**Eigenvalue spectrum panel:**
- Bar chart of first 10 smallest eigenvalues
- Annotate spectral gap (λ₂ - λ₁)

**Metrics panel (text):**
- Trustworthiness score (`sklearn.manifold.trustworthiness`, k=15)
- Silhouette score (`sklearn.metrics.silhouette_score`)
- Number of connected components in the k-NN graph
- Spectral gap value
- Eigenvalue condition number (λ₁₀ / λ₂)

**Matplotlib style:** `seaborn-v0_8-whitegrid` or similar clean style.

## Implementation Notes

- The existing `tests/environment.yml` already includes all required Python packages (matplotlib, scikit-learn, umap-learn, scipy, numpy)
- Use `argparse` for CLI
- Add a `load_dataset(name)` function that returns `(X, labels, name)` — this will be extended by Tier 2 datasets later
- Add a `export_graph(graph, path)` function for the CSR export — keep it reusable
- Print progress to stdout (dataset name, phase, timing)
- Handle the case where output directory doesn't exist (create it)

## Acceptance Criteria

- [ ] `tests/visual_eval/output/` is gitignored
- [ ] `tests/visual_eval/README.md` exists with usage instructions
- [ ] Script runs successfully with `python tests/visual_eval/generate_umap_comparisons.py --phase baseline`
- [ ] All 5 Tier 1 datasets generate graph `.npz`, spectral `.npy`, final `.npy`, labels `.npy`, and baseline `.png`
- [ ] Graph `.npz` files use the exact format compatible with `load_sparse_csr_f32_u32` (f32 data, i32 indices/indptr/shape)
- [ ] Baseline plots show sensible spectral init and UMAP embeddings for all datasets
- [ ] `--dataset` flag works to run a single dataset

## Files Created/Modified

- **Create:** `tests/visual_eval/generate_umap_comparisons.py`
- **Create:** `tests/visual_eval/README.md`
- **Create:** `tests/visual_eval/output/.gitkeep`
- **Modify:** `.gitignore` (add `tests/visual_eval/output/` entry, keep `.gitkeep`)
